### PR TITLE
provider/openstack: Safe Security Group Delete

### LINKED ISSUE #1782: OpenStack compute_secgroup resources - fails destroy when in use
```ini
resource "openstack_compute_instance_v2" "blendle" {
  depends_on = [
    "openstack_networking_router_v2.blendle",
    "openstack_compute_secgroup_v2.blendle"
  ]

  security_groups = ["blendle"]
}
```

```ini
resource "openstack_compute_secgroup_v2" "blendle" {
  name = "blendle"
  description = "Security rules for all \"blendle\" nodes"
}
```

the above fails on destroy, because Terraform tries to remove the security group while the machine is still active (being destroyed).

### PR DESCRIPTION
**Pull request by jtopjian**
This commit enables security groups to be deleted in a safe way by
checking their state over a period of time.

This fixes occurrences when the API says the instance is deleted but
it is still in the process of being deleted by OpenStack and thus the
security group returns an error saying that there are still instances
attached to the group.

Fixes #1782

### CODE REVIEW

**Comment by jtopjian** [comment]
This passes all current acceptance tests on OpenStack Kilo. 

I'll leave this open for a day or so for comments.

File: builtin/providers/openstack/resource_openstack_compute_secgroup_v2.go [hunk 1]
```diff
@@ -210,10 +212,20 @@ func resourceComputeSecGroupV2Delete(d *schema.ResourceData, meta interface{}) e
 		return fmt.Errorf("Error creating OpenStack compute client: %s", err)
```
Comment (phinze) [inline builtin/providers/openstack/resource_openstack_compute_secgroup_v2.go]: Noticed this message should be `s/creating/deleting/` while reviewing the diff below.

**Comment by phinze** [comment]
LGTM

File: builtin/providers/openstack/resource_openstack_compute_secgroup_v2.go (hunk 1 above)
Comment (jtopjian) [inline builtin/providers/openstack/resource_openstack_compute_secgroup_v2.go]: Actually, "creating" is correct here - it's making a new OpenStack client connection in order to do the delete. :smile: 

However, I do have a slightly cleaned up version coming right up.

Comment (phinze) [inline builtin/providers/openstack/resource_openstack_compute_secgroup_v2.go]: Whoops! Yes. :coffee: :coffee: :coffee:

**Comment by jtopjian** [comment]
Just pushed a slightly modified version that adds an additional error check as well as takes advantage of `utils.CheckDeleted` that I forgot exists.

**Comment by phinze** [comment]
Solid - looks good. :+1: 

@jtopjian you can merge - generally we merge with the GitHub button and immediately go to the CHANGELOG and use the in-browser edit to add a line about the change. I usually just copy the title from the PR, paste it into the CHANGELOG, edit it a bit and click save. :rocket:

**Comment by jtopjian** [comment]
Awesome! Thanks!

**Pull request merged**